api/apps/api/src/modules/access-control/scenarios-acl/scenario-acl.service.ts

```diff
@@ -0,0 +1,107 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { intersection } from 'lodash';
+import { Not, Repository } from 'typeorm';
+import { Permit } from '../access-control.types';
+import { Roles } from '../role.api.entity';
+import { UsersScenariosApiEntity } from '@marxan-api/modules/access-control/scenarios-acl/entity/users-scenarios.api.entity';
+import { ScenarioAccessControl } from '@marxan-api/modules/access-control/scenarios-acl/scenario-access-control';
+
+@Injectable()
+export class ScenarioAclService implements ScenarioAccessControl {
+  private readonly canEditScenarioRoles = [
+    Roles.scenario_contributor,
+    Roles.scenario_owner,
+  ];
+  private readonly canViewScenarioRoles = [
+    Roles.scenario_viewer,
+    Roles.scenario_contributor,
+    Roles.scenario_owner,
+  ];
+  private readonly canDeleteScenarioRoles = [Roles.scenario_owner];
+  private readonly canCreateSolutionRoles = [
+    Roles.scenario_owner,
+    Roles.scenario_contributor,
+  ];
+
+  private async getRolesWithinScenarioForUser(
+    userId: string,
+    scenarioId: string,
+  ): Promise<Array<Roles>> {
+    const rolesToCheck = (
+      await this.roles.find({
+        where: {
+          scenarioId,
+          userId,
+        },
+        select: ['roleName'],
+      })
+    ).flatMap((role) => role.roleName);
+    return rolesToCheck;
+  }
+
+  private async doesUserHaveRole(
+    roles: Roles[],
+    rolesToCheck: Roles[],
+  ): Promise<Permit> {
+    return intersection(roles, rolesToCheck).length > 0;
+  }
+
+  constructor(
+    @InjectRepository(UsersScenariosApiEntity)
+    private readonly roles: Repository<UsersScenariosApiEntity>,
+  ) {}
+
+  async canCreateSolution(userId: string, scenarioId: string): Promise<Permit> {
```

Comment:
what does that capability mean? Solutions are never created by user 🤔

Reply:
This was included in the ACL Permissions matrix, I guess in the end the triggering of a solution being created would be only possible for Scenario_owners and Scenario_contributors I guess